CONTRIBUTING.md

```diff
@@ -17,14 +17,41 @@ In order to build the source code of the .NET APM Agent you need
 please find or open an issue about it first.
 Talk about what you would like to do.
 It may be that somebody is already working on it,
 or that there are particular issues that you should know about before implementing the change.
 
+We aim to maintain high code quality, therefore every PR goes through the same review process, no matter who opened the PR.
+
+### General advice
+
+Please make sure that your PR addresses a single issue and its size is reasonable to review. There are no hard rules, but please keep in mind that every line of your PR will be reviewed and it's not uncommon that a longer discussion evolves in case of a sensitive change.
+
+Therefore it's preferred to have multiple smaller PRs over a single big PR.
+
+This makes sure that obvious changes get merged quickly and don't get blocked by unrelated changes. Additionally, the repository will also have a useful and searchable history by doing so.
+
+Please do:
+- Try to get feedback as early as possible and prefer to ask questions in issues before you start submitting code.
+- Add a description to your PR which describes your intention and gives a high level summary about your changes.
+- Run all the tests from the `test` folder and make sure that all of them are green. See [Testing](###Testing).
+- In case of new code, make sure it's covered by at least 1 test.
+- make sure your IDE uses the `.editorconfig` from the repo and you follow our coding guidelines. See [Coding-guidelines](###Coding-guidelines).
+- Feel free to close a PR and create a new one in case you changed your mind, or found a better solution.
+- Feel free to fix typos.
+- Feel free to use the draft PR feature of GitHub, in case you would like to show some work in progress code and get feedback on it.
+
+Please don't:
+- Create a PR where you address multiple issues at once.
+- Create a giant PR with a huge change set. There is no hard rule, but if your change grows over 1000 lines, it's maybe worth thinking about making it self contained and submit it as a PR and address follow-up issues in a subsequent PR. (of course there can be exceptions)
+- Actively push code to a PR until you haven't received any feedback. Of course if you spot some minor things after you opened a PR, it's perfectly fine to push a small fix. But please don't do active work on a PR that haven't received any feedback. It's very possible that someone already looked at it and is about to write a detailed review. If you actively add new changes to a PR then the reviewer will have a hard time to provide up to date feedback. If you just want to show work-in-progress code, feel free to use the draft feature of github, or indicate in the title, that the work is not 100% done yet. Of course, once you have feedback on a PR, it's perfectly fine, or rather encouraged to start working collaboratively on the PR and push new changes and address issues and suggestions from the reviewer.
```

Comment:
> Actively push code to a PR until you haven't received any feedback

This is how I would phrase it in Russian and Hebrew but I think in English "until" is followed by a _positive_ condition so it should be

> Actively push code to a PR until you have received feedback

@bmorelli25 Am I correct here?

Reply:
Good catch @SergeyKleyman, you're correct.

Reply:
Interesting, none of my languages work this way.

Fixed it, thanks!